stdlib/3/email/message.pyi

```diff
@@ -36,16 +38,16 @@ class Message:
     def get_charset(self) -> _CharsetType: ...
     def __len__(self) -> int: ...
     def __contains__(self, name: str) -> bool: ...
-    def __getitem__(self, name: str) -> Optional[str]: ...
-    def __setitem__(self, name: str, val: str) -> None: ...
+    def __getitem__(self, name: str) -> Optional[_HeaderType]: ...
```

Comment:
I've had a look at that.

My concern is that a header is _very_ different to a string: .lower(), .strip() etc all completely fail on a Header. There is little you can correctly do with the result of get/get_all/getitem without either checking the type, running str() on it, or passing it to another function that handles either type. Otherwise you *will* eventually hit type errors. This hit us at the Patchwork project when a header unexpectedly contained an invalid byte and our parser broke completely.

I'm happy to look at another way to approach the problem though - this is the first time I've touched typeshed.